itest/assets_test.go

```diff
@@ -673,6 +683,86 @@ func testAssetBalances(t *harnessTest) {
 	recipients := map[string]uint64{
 		bobAddr.Encoded: bobAddr.Amount,
 	}
+	out, err := wire.NewOutPointFromString(
+		targetAsset.ChainAnchor.AnchorOutpoint,
+	)
+	require.NoError(t.t, err)
+
+	// 1. Fail if Inputs are provided but Outpoint is nil.
+	emptyOutpointPrevId := &wrpc.PrevId{
+		Outpoint:  nil,
+		Id:        assetId,
+		ScriptKey: scriptKey,
+	}
+	_, err = aliceTapd.FundVirtualPsbt(
+		ctxt, &wrpc.FundVirtualPsbtRequest{
+			Template: &wrpc.FundVirtualPsbtRequest_Raw{
+				Raw: &wrpc.TxTemplate{
+					Recipients: recipients,
+					Inputs: []*wrpc.PrevId{
+						emptyOutpointPrevId,
+					},
+				},
+			},
+		},
+	)
+	require.Error(t.t, err)
+	require.Contains(t.t, err.Error(), "index 0 has a nil Outpoint")
+
+	// 2. Fail if Inputs contain an invalid Outpoint.
+	invalidTxidPrevId := &wrpc.PrevId{
```

Comment:
Could convert all of this into a table driven test (and then add some test cases for too short asset ID and script key):

```
subTests := []struct {
		name           string
		inputs         []*wrpc.PrevId
		expectError    bool
		expectedErrMsg string
	}{
		{
			name: "Fail if Inputs are provided but Outpoint is nil",
			inputs: []*wrpc.PrevId{
				{
					Outpoint:  nil,
					Id:        assetId,
					ScriptKey: scriptKey,
				},
			},
			expectError:    true,
			expectedErrMsg: "index 0 has a nil Outpoint",
		},
		{
			name: "Fail if Inputs contain an invalid Outpoint",
			inputs: []*wrpc.PrevId{
				{
					Outpoint: &taprpc.OutPoint{
						Txid: []byte(
							"invalid_txid",
						),
						OutputIndex: 0,
					},
					Id:        assetId,
					ScriptKey: scriptKey,
				},
			},
			expectError:    true,
			expectedErrMsg: "invalid Txid length",
		},
		{
			name: "Succeed if a valid Outpoint is provided",
			inputs: []*wrpc.PrevId{
				{
					Outpoint: &taprpc.OutPoint{
						Txid:        out.Hash[:],
						OutputIndex: out.Index,
					},
					Id:        assetId,
					ScriptKey: scriptKey,
				},
			},
			expectError: false,
		},
	}

	for _, tt := range subTests {
		_, err := aliceTapd.FundVirtualPsbt(
			ctxt, &wrpc.FundVirtualPsbtRequest{
				Template: &wrpc.FundVirtualPsbtRequest_Raw{
					Raw: &wrpc.TxTemplate{
						Recipients: recipients,
						Inputs:     tt.inputs,
					},
				},
			},
		)

		if tt.expectError {
			require.ErrorContains(t.t, err, tt.expectedErrMsg)
		} else {
			require.NoError(t.t, err)

			// Unlock the input so that it can be spent again.
			for _, input := range tt.inputs {
				_, err = aliceTapd.RemoveUTXOLease(
					ctxb, &wrpc.RemoveUTXOLeaseRequest{
						Outpoint: input.Outpoint,
					},
				)
				require.NoError(t.t, err)
			}
		}
	}
```